scRNA-seq-advanced/setup/intro-setup.Rmd

```diff
@@ -0,0 +1,340 @@
+
+```
+
+### Behold, the filtered and normalized SCE:
+
+```{r}
+norm_sce
+
+altExp(norm_sce)
+```
+
+## Dimension reduction
+
+HVGs:
+```{r}
+num_genes <- 5000 # 2000? 
+
+gene_variance <- scran::modelGeneVar(norm_sce)
+hvg_list <- scran::getTopHVGs(gene_variance,
+                               n = num_genes)
+```
+
+PCA:
+```{r}
+# Do we want to set the number of PCs?
+norm_sce <- scater::runPCA(norm_sce,
+                           subset_row = hvg_list)
+```
+
+UMAP:
+```{r I map you map we all map for MapMap}
+# Perform graph clustering first
+nn_k <- 15
+nn_clusters <- clusterRows(
+  reducedDim(norm_sce, "PCA"), 
+  NNGraphParam(k = nn_k, 
+               cluster.fun = "walktrap")
+)
+
+# pop back into sce
+norm_sce$nn_clusters <- factor(nn_clusters)
+
+# Add in UMAPs and plot with colors
+norm_sce <- scater::runUMAP(norm_sce,
+                            dimred = "PCA")
+scater::plotReducedDim(norm_sce, "UMAP", colour_by = "nn_clusters")
+```
+
+## Cell type annotation
+
+We can introduce annotation with `SingleR`, and it might be a good idea to show how the results are highly influenced by the reference. 
+For now, I'm just using the `HumanPrimaryCellAtlasData` reference as a starting place.
+
+
+```{r}
+ref_data <- celldex::HumanPrimaryCellAtlasData(ensembl = TRUE) # pick one for now to start
+preds <- SingleR::SingleR(
+  test = norm_sce, 
+  ref = ref_data,
+  labels = ref_data$label.main # "coarse-grained" labels
+)
+preds
+
+# We can place these into the sce
+norm_sce$celltype_predictions <- preds$pruned.labels
+
+# **heavily** biased towards monocytes and T-cells
+celltype_counts <- table(preds$pruned.labels) %>%
+  as.data.frame() %>%
+  dplyr::rename(
+    celltype = Var1, 
+    n = Freq
+  ) %>%
+  dplyr::arrange(-n)
+
+celltype_counts
+
+```
+
+
+
+Visualize the predictions in a couple ways:
+```{r}
+SingleR::plotScoreHeatmap(preds)
+```
+
+
+```{r UMAP for the people}
+# Not too shabby!
+scater::plotReducedDim(norm_sce, "UMAP", colour_by = "celltype_predictions", 
+                       point_alpha = 0.4, point_size = 0.25)
+```
+
+
+It might be helpful to hone in on the most prevalent celltypes only for ease of visualization.
+
+```{r}
+# The top 5 cell types account for over 98% of the cells!
+celltype_counts %>%
+  dplyr::slice(1:5) %>%
+  dplyr::summarize(sum(n) / sum(celltype_counts$n))
+
+top_celltypes <- as.character(celltype_counts$celltype[1:5]) 
+
+# Recode a new celltype vector with top 5 separated from "other" using some forcats magic
+norm_sce$celltype_predictions_top <- forcats::fct_lump_n(norm_sce$celltype_predictions, 5)
+
+scater::plotReducedDim(norm_sce, "UMAP", colour_by = "celltype_predictions_top", 
+                       point_alpha = 0.6, point_size = 0.25)
+```
+
+## ADT "celltyping"
+
+We can leverage the ADTs as protein markers to identify likely celltypes, which can be compared back to the `SingleR` annotations as a form of validation.
```

Comment:
For pedagogy reasons, I think we will do this the other way around. Antibody tags are the "traditional" way of identifying cell types, so do that first, then show that we can also do it with only RNA.